### PR TITLE
Added url length assessment

### LINKED ISSUE #338: Create assessment for url length
Create assessment for url length

### PR DESCRIPTION
**Pull request by diedexx**
Fixes #338

### CODE REVIEW

File: js/analyses/isUrlTooLong.js [hunk 1]
```diff
@@ -5,16 +5,15 @@ var analyzerConfig = require( "../config/analyzerConfig" )();
...
- * @param {string} url The URL to check the length from.
- * @param {string} keyword The keyword
+ * @param {object} paper the paper to run this assessment on
  * @returns {boolean} true if the URL is too long, false if it isn't
  */
-module.exports = function( url, keyword ) {
-	var urlLength = url.length;
-	var keywordLength = keyword.length;
-	var isUrlTooLong = false;
+module.exports = function( paper ) {
```
Comment (jcomack) [inline js/analyses/isUrlTooLong.js]: Please consider renaming this file. `isUrlTooLong` suggests a yes or no 'response', whereas now it's a true or false response.

Possibly renaming this to `urlIsTooLong` would already be an improvement.

File: js/analyses/isUrlTooLong.js [hunk 2]
```diff
@@ -5,16 +5,15 @@ var analyzerConfig = require( "../config/analyzerConfig" )();
...
+	var urlLength = paper.getUrl().length;
+	var keywordLength = paper.getKeyword().length;
+
 	if ( urlLength > analyzerConfig.maxUrlLength  && urlLength > keywordLength + analyzerConfig.maxSlugLength ) {
-		isUrlTooLong = true;
+		return true;
```
Comment (jcomack) [inline js/analyses/isUrlTooLong.js]: Please remove the reference to the analyzerConfig and set the values defined in the config, in the research.

File: js/analyzer.js [hunk 3]
```diff
@@ -401,12 +401,7 @@ Analyzer.prototype.urlKeyword = function() {
  * @returns {{test: string, result: number}[]}
  */
 Analyzer.prototype.urlLength = function() {
-	var result = [ { test: "urlLength", result: { urlTooLong: isUrlTooLong(
-		this.paper.getUrl(),
-		this.paper.getKeyword(),
-		this.config.maxSlugLength,
-		this.config.maxUrlLength
-	) } } ];
+	var result = [ { test: "urlLength", result: { urlTooLong: isUrlTooLong( this.paper ) } } ];
```
Comment (jcomack) [inline js/analyzer.js]: Unsure if this method is necessary anymore. @terw-dan, can you confirm this?

File: js/analyzescorer.js [hunk 4]
```diff
@@ -9,6 +9,7 @@ var isUndefined = require( "lodash/lang/isUndefined" );
 
 var assessments = {};
 assessments.wordCount = require( "./assessments/countWords.js" );
+assessments.urlLength = require( "./assessments/urlLength.js" );
```
Comment (jcomack) [inline js/analyzescorer.js]: Consider renaming this. We're not assessing the length of the url; we're assessing whether or not the url is too long or not. Something like `urlIsTooLong` would be a better fit.

File: js/assessments/urlLength.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * The assessment that checks the url length
+ *
+ * @param {Paper} paper The paper to run this assessment on.
+ * @param {object} researcher The researcher used for the assessment.
+ * @param {object} i18n The i18n-object used for parsing translations.
+ * @returns {object} an AssessmentResult with the score and the formatted text.
+ */
+var urlLengthAssessment = function( paper, researcher, i18n ) {
```
Comment (jcomack) [inline js/assessments/urlLength.js]: Consider renaming this. We're not assessing the length of the url; we're assessing whether or not the url is too long or not. Something like `urlIsTooLongAssessment` would be a better fit.

File: spec/assessments/urlLengthAssessmentSpec.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+var urlLengthAssessment = require( "../../js/assessments/urlLength.js" );
```
Comment (jcomack) [inline spec/assessments/urlLengthAssessmentSpec.js]: Same as with the assessment name: please consider renaming it.

**Comment by jcomack** [comment]
CR done

**Comment by atimmer** [comment]
I would say that `isUrlTooLong` is not a research because it is not objective. We should move the check if the URL is too long to the assessment.

File: js/analyzer.js (hunk 3 above)
Comment (terw-dan) [inline js/analyzer.js]: This will be indeed deprecated, the analyzer will be removed if the assessments are finished.

**Comment by jcomack** [comment]
CR done :+1:

**Comment by jcomack** [comment]
Acceptance done :+1:

**Pull request merged**